internals-js/src/coreSpec.ts

```diff
@@ -768,6 +774,10 @@ export function removeAllCoreFeatures(schema: Schema) {
   const errors: GraphQLError[] = [];
   for (const { feature, type, references } of typeReferences) {
     const referencesInSchema = references.filter(r => r.isAttached());
+    const referencedFeature = referencesInSchema.find(r => directivesToPromote.includes(`@${r.parent.name}`));
```

Comment:
Yeah, so the issue here was that I was getting an error message trying to expose `@link` that I thought wasn't very helpful.
`The supergraph schema failed to produce a valid API schema`

As you rightly point out, it's because of the `link__Purpose` and `link__Import` references, and so I decided it would be safest to not allow directives that referenced the "core types" to be exposed, hoping that if we didn't allow promotion of directives with dangling core types, that would be safest for the general case. Happy to go in whichever direction you think is best.